implementing-nosto/implement-search/implement-search-using-code-editor/implementing-autocomplete.md

```diff
@@ -189,6 +189,41 @@ export default () => {
 ```
 {% endcode %}
 
+
+#### Voice to text search
+
+To implement voice to text search in search templates, additional configuration params needs to be sent:
+
+{% code title="index.js" %}
+```javascript
+import { init } from '@nosto/preact'
+
+import speechToTextComponent from "./SpeechToTextComponent"
+
+init({
+    ...window.nostoTemplatesConfig,
+    speechToTextComponent,
+    speechToTextEnabled: true,
+    ...
+})
+```
+{% endcode %}
+
+##### Configuration Parameters:
+
+- **speechToTextComponent** – The component that renders the voice search button.
+- **speechToTextEnabled** – A flag to enable or disable the voice search feature.
+
+The voice search button will be injected adjacent to the search input field, positioned as an overlay on the right end of the input.
+
+Within the button component, the `useSpeechToText` hook is used to toggle voice input on and off.
+
+The `@nosto/preact` package exports two useful utilities:
+- **useSpeechToText** – A hook to control the voice-to-text functionality.
+- **speechToTextSupported** – A variable indicating whether the current environment supports the feature.
+
+>Note: Some additional styling may be required depending on the merchant’s input setup to ensure proper alignment and positioning of the voice search button.
```

Comment:
Not sure if this is needed. The necessity of custom styling is implied in Search Templates usage